src/JSInterop/Microsoft.JSInterop/src/DotNetObjectReferenceOfT.cs

```diff
@@ -50,10 +47,36 @@ internal long ObjectId
             get
             {
                 ThrowIfDisposed();
+                Debug.Assert(_objectId != 0, "Accessing ObjectId without tracking is always incorrect.");
+
                 return _objectId;
             }
         }
 
+        internal long TrackObjectReference(JSRuntime jsRuntime)
```

Comment:
Does it make sense to change this pattern? This method calls `jsRuntime.TrackObjectReference(this)`. It's a little odd to have the visitor pattern in use without polymorphism. 

So what I had in mind would be something like making the JSRuntime+ObjectId internal set properties, and then moving all of this logic to the JSRuntime class.